pkg/webhook/webhook.go

```diff
@@ -386,3 +393,22 @@ func genCertAndKeyFile(certData, keyData []byte, certDir string) error {
 	klog.V(2).InfoS("successfully generate certificate and key files")
 	return nil
 }
+
+// bindWebhookConfigToFleetSystem sets the OwnerReference of the argued ValidatingWebhookConfiguration to the cluster scoped fleet-system namespace.
+func bindWebhookConfigToFleetSystem(ctx context.Context, k8Client client.Client, validatingWebhookConfig *admv1.ValidatingWebhookConfiguration) error {
+	var fleetNs corev1.Namespace
+	if err := k8Client.Get(ctx, client.ObjectKey{Name: "fleet-system"}, &fleetNs); err != nil {
```

Comment:
What if we change the fleet system namespace name?

Reply:
I agree that having this as a string literal is a problem. A constant won't matter much either. We want to bind to the namespace that the components reside within. This is mastered within the helm chart. 

Ideas:

1. Add an option for the hub-agent. Allowing the value to be passed in from the helm chart values.
2. Utilize the helm hook binary execution approach.

Reply:
how does the second approach work?

Reply:
We would develop a Go binary that would packaged in a cleanup image. The image would would be ran via  Kubernetes Job. The job manifest would contain the Helm Hook annotations required for Helm to determine where in the release lifecycle to intervene with the Job e.g. pre-delete.  The annotations would also specify how to cleanup the resourced (Job) created via this hook.

Something akin to:
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: fleet-uninstall-cleanup-job
  annotations:
    "helm.sh/hook": pre-delete
    "helm.sh/hook-weight": "0"
    "helm.sh/hook-delete-policy": hook-succeeded
spec:
  template:
    metadata:
      name: {{ .Release.Name }}-nginx
      labels:
        {{- include "nginx.labels" . | indent 8 }}
    spec:
      restartPolicy: Never
      containers:
      - name: fleet-uninstall-cleanup-job
        image: "hub-uninstall-cleanup:1.0"
```